includes/digilan-connection.php

```diff
@@ -688,6 +709,24 @@ private static function get_connection_with_mac($mac)
         return false;
     }
 
+    private static function get_hostname_with_ap_mac($mac)
+    {
+        $settings = clone DigilanToken::$settings;
+        $access_points = $settings->get('access-points');    
+        foreach ($access_points as $key=>$value) {
+            $global_settings = array(
+                'mac' => $access_points[$key]['mac']
+            );
+            $current_ap_setting = $access_points[$key]['specific_ap_settings'];
+            $global_settings = $current_ap_setting->get_ap_params($global_settings);
+
+            if ($mac == $global_settings['mac']) {
+                return $key;
+            }
+        }
+        return false;
+    }
```

Comment:
comprend pas , les anciens settings semble ignoré et le nom de la fonction est pas comprehensible cela retourne un KEY  et le truc s appelle get_hostname....

Reply:
faut renommer KEY en hostname...

Reply:
et appeler get_hostname_from_mac
avec kle commenatire , try to find a specific setting for a mac , and return the hostname 
( si c est ce que cela fait )